lib/ansible/plugins/lookup/hashi_vault.py

```diff
@@ -0,0 +1,93 @@
+# (c) 2015, Jonathan Davila <jdavila(at)ansible.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+# USAGE: {{ lookup('hashi_vault', 'secret=secret/hello token=c975b780-d1be-8016-866b-01d0f9b688a5 url=http://myvault:8200')}}
+#
+# You can skip setting the url if you set the VAULT_ADDR environment variable
+# or if you want it to default to localhost:8200
+#
+# NOTE: Due to a current limitation in the HVAC library there won't
+# necessarily be an error if a bad endpoint is specified.
+#
+# Requires hvac library. Install with pip.
+#
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+
+from ansible.errors import *
+from ansible.plugins.lookup import LookupBase
+
+try:
```

Comment:
this should be moved into the run method, as we only want this to error on running, not on loading, which will always happen if this is included in ansible.